include/gridtools/common/int_vector.hpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * Implements operations on `int_vector`s, which are hymaps of integers (integral types or integral_constants)
+ *
+ * A type `T` is `int_vector` if
+ *  - `is_hymap<T>`
+ *  - all value types `V` are std::is_integral<V> or is_gr_integral_constant<V>
+ *  - `std::is_trivially_copy_constructible<T>` // TODO isn't that trivially satisfied by the previous conditions?
+ */
+
+#pragma once
+
+#include <type_traits>
+
+#include "../meta.hpp"
+#include "defs.hpp"
+#include "gridtools/meta/debug.hpp"
+#include "host_device.hpp"
+#include "hymap.hpp"
+#include "integral_constant.hpp"
+#include "tuple.hpp"
+#include "tuple_util.hpp"
+
+namespace gridtools {
+
+    namespace int_vector_impl_ {
+        template <class... Ts>
+        struct value_t_merger;
+
+        template <template <class...> class Item, class Key, class... Vs>
+        struct value_t_merger<Item<Key, Vs>...> {
+            using type = Item<Key, std::decay_t<decltype((Vs{} + ...))>>;
+        };
+
+        template <class... T>
+        using merger_t = typename value_t_merger<T...>::type;
+
+        template <class Key>
+        struct add_f {
+            template <class... Ts>
+            GT_FUNCTION GT_CONSTEXPR decltype(auto) operator()(Ts const &...args) const {
+                return (host_device::at_key_with_default<Key, integral_constant<int, 0>>(args) + ...);
+            }
+        };
+
+        /**
+         * @brief Returns elementwise sum of `int_vector`s
+         *
+         * The keys of the resulting `int_vector` are the union of the keys of the operands.
+         */
+        template <class First, class Second>
+        GT_FUNCTION GT_CONSTEXPR auto plus(First &&first, Second &&second) {
+            using merged_meta_map_t = meta::mp_make<int_vector_impl_::merger_t,
+                meta::concat<hymap::to_meta_map<First>, hymap::to_meta_map<Second>>>;
+            using keys_t = meta::transform<meta::first, merged_meta_map_t>;
+            using generators = meta::transform<add_f, keys_t>;
+            return tuple_util::host_device::generate<generators, hymap::from_meta_map<merged_meta_map_t>>(
+                std::forward<First>(first), std::forward<Second>(second));
```

Comment:
`wstd::forward` here and below 
+ add the appropriate header

Reply:
done